src/transformers/models/bloom/configuration_bloom.py

```diff
@@ -0,0 +1,170 @@
+        skip_bias_add_qkv (`bool`, *optional*, defaults to `False`):
+            If set to `True`, it will skip bias add for the first linear layer in the transformer blocks
+        attention_softmax_in_fp32 (`bool`, *optional*, defaults to `True`):
+            If set to `True` and the `dtype` is set to `float16` it will scale the input of the Softmax function to
+            `fp32`
+        hidden_dropout (`float`, *optional*, defaults to 0.1):
+            Dropout rate of the dropout function in `bias_dropout_fusion`
+        attention_dropout (`float`, *optional*, defaults to 0.1):
+            Dropout rate applied to the attention probs
+        scale_attn_weights (`bool`, *optional*, defaults to `True`):
+            Scale attention weights by dividing by sqrt(hidden_size)..
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models).
+        scale_attn_by_inverse_layer_idx (`bool`, *optional*, defaults to `False`):
+            Whether to additionally scale attention weights by `1 / layer_idx + 1`.
+        reorder_and_upcast_attn (`bool`, *optional*, defaults to `False`):
+            Whether to scale keys (K) prior to computing attention (dot-product) and upcast attention
+            dot-product/softmax to float() when training with mixed precision.
+
+    Example:
+
+    ```python
+    >>> from transformers import BloomModel, BloomConfig
+
+    >>> # Initializing a Bloom configuration
+    >>> configuration = BloomConfig()
+
+    >>> # Initializing a model from the configuration
+    >>> model = BloomModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "bloom"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {
+        "max_position_embeddings": "seq_length",
+        "num_hidden_layers": "n_layer",
+        "n_head": "num_attention_heads",
+        "hidden_size": "n_embed",
+    }
+
+    def __init__(
+        self,
+        vocab_size=250880,
+        seq_length=20,  # This constraints alibi embedding size together with offset_alibi
+        offset_alibi=100,
+        hidden_size=64,
+        n_layer=2,
+        n_head=8,
+        n_inner=None,
+        masked_softmax_fusion=True,
+        layer_norm_epsilon=1e-5,
+        initializer_range=0.02,
+        use_cache=False,
+        bos_token_id=50256,
+        eos_token_id=50256,
+        apply_residual_connection_post_layernorm=False,
+        bias_dropout_fusion=True,
+        skip_bias_add=True,
+        skip_bias_add_qkv=False,
+        hidden_dropout=0.0,
+        attention_dropout=0.0,
+        attention_softmax_in_fp32=True,
+        pretraining_tp=1,  # TP rank used when training with megatron
+        pretraining_pp=1,  # PP rank used when training with megatron
```

Comment:
What are those? Are they necessary? They're not mentioned in the docs. Can the model be used independently of megatron?

Reply:
I wonder if the model should have arguments to run on a specific framework like this.

Reply:
Actually these are quite important. We can probably skip the `pretraining_p` but the `pretraining_tp` is important to reproduce the same operations as in Megatron. 
For more details you can see: https://github.com/pytorch/pytorch/issues/76232#event-6491299095

Reply:
I will mention them in the doc in a next commit :-)

Reply:
Updated it here: 9a017ffb5ba9e0ac25f12040abee142fca4988ba

Reply:
Just removed the `pretraining_pp` on the config files of the models on the Hub + config file of the repo!